### ISSUE TITLE
"Control Right" remote button press causes exception on android

### ISSUE THREAD

**Issue by RtypeStudios** [flutter/flutter#118239]
## Steps to Reproduce

Using Flutter 3.3.10 create a new application:

1. Create a fresh application using "flutter create app"
2. Using the remote on [Mediabox Maverick 4K Android TV Box](https://www.takealot.com/mediabox-maverick-4k-android-tv-box-netflix-google-certified-dis/PLID71641421) in IR mode try to click the "+" button. 

**Expected results:**
For the counter in the centre of the screen to increment by one.

**Actual results:**
No UI changes and an exception is thrown:
```
════════ Exception caught by services library ══════════════════════════════════
The following assertion was thrown during a platform message callback:
Attempted to send a key down event when no keys are in keysPressed. This state can occur if the key event being sent doesn't properly set its modifier flags. This was the event: RawKeyDownEvent#2293d(logicalKey: LogicalKeyboardKey#0050c(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select"), physicalKey: PhysicalKeyboardKey#700e4(usbHidUsage: "0x000700e4", debugName: "Control Right"), repeat: false) and its data: RawKeyEventDataAndroid#955db(flags: 8, codePoint: 0, plainCodePoint: 0, keyCode: 23, scanCode: 97, metaState: 0)
'package:flutter/src/services/raw_keyboard.dart':
package:flutter/…/services/raw_keyboard.dart:1
Failed assertion: line 682 pos 7: 'event is! RawKeyDownEvent || _keysPressed.isNotEmpty'

2

Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack
#2      RawKeyboard.handleRawKeyEvent
package:flutter/…/services/raw_keyboard.dart:682
#3      KeyEventManager.handleRawKeyMessage
package:flutter/…/services/hardware_keyboard.dart:878
#4      BasicMessageChannel.setMessageHandler.<anonymous closure>
package:flutter/…/services/platform_channel.dart:197
#5      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure>
package:flutter/…/services/binding.dart:387
#6      _invoke2 (dart:ui/hooks.dart:186:13)
#7      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
#8      _Channel.push (dart:ui/channel_buffers.dart:132:31)
#9      ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
#10     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:599:22)
#11     _dispatchPlatformMessage (dart:ui/hooks.dart:89:31)
(elided 2 frames from class _AssertionError)
════════════════════════════════════════════════════════════════════════════════

════════ Exception caught by services library ══════════════════════════════════
A KeyUpEvent is dispatched, but the state shows that the physical key is pressed on a different logical key. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#3eaf7(physicalKey: PhysicalKeyboardKey#700e4(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#00101(keyId: "0x200000101", keyLabel: "Control Right", debugName: "Control Right"), character: null, timeStamp: 0:00:16.300340, synthesized) and the recorded logical key LogicalKeyboardKey#0050c(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select")
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 435 pos 16: '_pressedKeys[event.physicalKey] == event.logicalKey'

════════════════════════════════════════════════════════════════════════════════

```

<details>
<summary>Code sample</summary>
This occurs with the sample created using the command:
`flutter create test`
</details>

<details>
  <summary>Logs</summary>

[out.txt](https://github.com/flutter/flutter/files/10382688/startup-out.txt)
[out-with-error.txt](https://github.com/flutter/flutter/files/10382701/out-with-error.txt)

```                                 
Analyzing app...
No issues found! (ran in 3.5s)
```

```
[√] Flutter (Channel stable, 3.3.10, on Microsoft Windows [Version 10.0.19045.2364], locale en-AU)
    • Flutter version 3.3.10 on channel stable at C:\Users\Alex\fvm\versions\3.3.10
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (4 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Program Files (x86)\Android\android-sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Program Files\Microsoft\jdk-11.0.12.7-hotspot\bin\java
    • Java version OpenJDK Runtime Environment Microsoft-25199 (build 11.0.12+7)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.4.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33205.214
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.74.2)
    • VS Code at C:\Users\Alex\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.56.0

[√] Connected device (5 available)
    • SM T510 (mobile)        • R52N10VZK6B       • android-arm    • Android 9 (API 28)
    • MBX4K Maverick (mobile) • 192.168.2.54:5555 • android-arm    • Android 10 (API 29)
    • Windows (desktop)       • windows           • windows-x64    • Microsoft Windows [Version 10.0.19045.2364]
    • Chrome (web)            • chrome            • web-javascript • Google Chrome 108.0.5359.125
    • Edge (web)              • edge              • web-javascript • Microsoft Edge 108.0.1462.76

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
@RtypeStudios 
Can you try the same on latest master version and see if the issue persists ?

`flutter channel master`
`flutter upgrade`

**Comment by RtypeStudios** [comment]
@darshankawar Thanks for the suggestion.

Sadly same error:

`An Observatory debugger and profiler on MBX4K Maverick is available at: http://127.0.0.1:59863/RYVHw2LJVYM=/
E/ion     ( 8042): ioctl c0044901 failed with code -1: Invalid argument
W/gralloc ( 8042): WARNING: internal format modifier bits not mutually exclusive. AFBC basic bit is always set, so extended AFBC support bits must always be checked.
The Flutter DevTools debugger and profiler on MBX4K Maverick is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:59863/RYVHw2LJVYM=/
E/flutter ( 8042): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/services/hardware_keyboard.dart': Failed assertion: line 437 pos 16: '_pressedKeys[event.physicalKey] == event.logicalKey': A KeyUpEvent is dispatched, but the state shows that the physical key is pressed on a different logical key. If this occurs in real application, please report this bug 
to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#aa336(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#d24f3(keyId: "0x200000101", keyLabel: "Control Right", debugName: "Control Right"), character: null, timeStamp: 0:00:20.496932, synthesized) and the recorded logical key LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select")
E/flutter ( 8042): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter ( 8042): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter ( 8042): #2      HardwareKeyboard._assertEventIsRegular.<anonymous closure> (package:flutter/src/services/hardware_keyboard.dart:437:16)
E/flutter ( 8042): #3      HardwareKeyboard._assertEventIsRegular (package:flutter/src/services/hardware_keyboard.dart:445:6)
E/flutter ( 8042): #4      HardwareKeyboard.handleKeyEvent (package:flutter/src/services/hardware_keyboard.dart:537:5)
E/flutter ( 8042): #5      KeyEventManager.handleKeyData (package:flutter/src/services/hardware_keyboard.dart:883:29)
E/flutter ( 8042): #6      PlatformDispatcher._keyDataListener.<anonymous closure>.<anonymous closure> (dart:ui/platform_dispatcher.dart:430:41)
E/flutter ( 8042): #7      _invoke1 (dart:ui/hooks.dart:164:13)
E/flutter ( 8042): #8      PlatformDispatcher._keyDataListener.<anonymous closure> (dart:ui/platform_dispatcher.dart:428:7)
E/flutter ( 8042): #9      _invoke2 (dart:ui/hooks.dart:183:13)
E/flutter ( 8042): #10     _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:40:5)
E/flutter ( 8042): #11     _Channel.push (dart:ui/channel_buffers.dart:130:31)
E/flutter ( 8042): #12     ChannelBuffers.push (dart:ui/channel_buffers.dart:326:17)
E/flutter ( 8042): #13     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:664:22)
E/flutter ( 8042): #14     _dispatchPlatformMessage (dart:ui/hooks.dart:86:31)
E/flutter ( 8042):
E/flutter ( 8042): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/services/hardware_keyboard.dart': Failed assertion: line 437 pos 16: '_pressedKeys[event.physicalKey] == event.logicalKey': A KeyUpEvent is dispatched, but the state shows that the physical key is pressed on a different logical key. If this occurs in real application, please report this bug 
to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyUpEvent#aa336(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#d24f3(keyId: "0x200000101", keyLabel: "Control Right", debugName: "Control Right"), character: null, timeStamp: 0:00:20.496932, synthesized) and the recorded logical key LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select")
E/flutter ( 8042): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter ( 8042): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter ( 8042): #2      HardwareKeyboard._assertEventIsRegular.<anonymous closure> (package:flutter/src/services/hardware_keyboard.dart:437:16)
E/flutter ( 8042): #3      HardwareKeyboard._assertEventIsRegular (package:flutter/src/services/hardware_keyboard.dart:445:6)
E/flutter ( 8042): #4      HardwareKeyboard.handleKeyEvent (package:flutter/src/services/hardware_keyboard.dart:537:5)
E/flutter ( 8042): #5      KeyEventManager.handleKeyData (package:flutter/src/services/hardware_keyboard.dart:883:29)
E/flutter ( 8042): #6      PlatformDispatcher._keyDataListener.<anonymous closure>.<anonymous closure> (dart:ui/platform_dispatcher.dart:430:41)
E/flutter ( 8042): #7      _invoke1 (dart:ui/hooks.dart:164:13)
E/flutter ( 8042): #8      PlatformDispatcher._keyDataListener.<anonymous closure> (dart:ui/platform_dispatcher.dart:428:7)
E/flutter ( 8042): #9      _invoke2 (dart:ui/hooks.dart:183:13)
E/flutter ( 8042): #10     _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:40:5)
E/flutter ( 8042): #11     _Channel.push (dart:ui/channel_buffers.dart:130:31)
E/flutter ( 8042): #12     ChannelBuffers.push (dart:ui/channel_buffers.dart:326:17)
E/flutter ( 8042): #13     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:664:22)
E/flutter ( 8042): #14     _dispatchPlatformMessage (dart:ui/hooks.dart:86:31)
E/flutter ( 8042):

══╡ EXCEPTION CAUGHT BY SERVICES LIBRARY ╞══════════════════════════════════════════════════════════
The following assertion was thrown during a platform message callback:
A KeyDownEvent is dispatched, but the state shows that the physical key is already pressed. If this
occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please
ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This
was the event: KeyDownEvent#33d12(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4",
debugName: "Control Right"), logicalKey: LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel:
"Select", debugName: "Select"), character: null, timeStamp: 0:00:20.499312)
'package:flutter/src/services/hardware_keyboard.dart':
Failed assertion: line 430 pos 16: '!_pressedKeys.containsKey(event.physicalKey)'

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack:
#2      HardwareKeyboard._assertEventIsRegular.<anonymous closure> (package:flutter/src/services/hardware_keyboard.dart:430:16)
#3      HardwareKeyboard._assertEventIsRegular (package:flutter/src/services/hardware_keyboard.dart:445:6)
#4      HardwareKeyboard.handleKeyEvent (package:flutter/src/services/hardware_keyboard.dart:537:5)
#5      KeyEventManager.handleRawKeyMessage (package:flutter/src/services/hardware_keyboard.dart:956:37)
#6      BasicMessageChannel.setMessageHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:214:49)
#7      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure> (package:flutter/src/services/binding.dart:393:35)
#8      _invoke2 (dart:ui/hooks.dart:183:13)
#9      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:40:5)
#10     _Channel.push (dart:ui/channel_buffers.dart:130:31)
#11     ChannelBuffers.push (dart:ui/channel_buffers.dart:326:17)
#12     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:664:22)
#13     _dispatchPlatformMessage (dart:ui/hooks.dart:86:31)
(elided 2 frames from class _AssertionError)
════════════════════════════════════════════════════════════════════════════════════════════════════

Another exception was thrown: A KeyDownEvent is dispatched, but the state shows that the physical key is already pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyDownEvent#33d12(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select"), character: null, timeStamp: 0:00:20.499312)   

Another exception was thrown: A KeyDownEvent is dispatched, but the state shows that the physical key is already pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyDownEvent#33d12(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select"), character: null, timeStamp: 0:00:20.499312)   

Another exception was thrown: A KeyDownEvent is dispatched, but the state shows that the physical key is already pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyDownEvent#33d12(physicalKey: PhysicalKeyboardKey#f5c25(usbHidUsage: "0x000700e4", debugName: "Control Right"), logicalKey: LogicalKeyboardKey#4999d(keyId: "0x10000050c", keyLabel: "Select", debugName: "Select"), character: null, timeStamp: 0:00:20.499312)   
`

**Comment by darshankawar** [comment]
Thanks for the update. I think this is similar to https://github.com/flutter/flutter/issues/101275, so please follow-up in it for further updates. You can try workaround mentioned in this comment to see if it works in your case.
Closing this as a duplicate.
